controllers/operatorconfigmap_controller.go

```diff
@@ -279,35 +285,28 @@ func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	// ownerReference on it as its cluster scoped resource
 	cephfsCSIDriver := templates.CephFSCSIDriver.DeepCopy()
 	cephfsCSIDriver.ObjectMeta.Name = csi.GetCephFSDriverName()
-	err = csi.CreateCSIDriver(c.ctx, c.Client, cephfsCSIDriver)
-	if err != nil {
+	if err := csi.CreateCSIDriver(c.ctx, c.Client, cephfsCSIDriver); err != nil {
```

Comment:
@bernerhat @Madhu-1 I think something is missing here. 
I don't see a parameter that will allow us to add an owner ref on the CSI driver deployment! 
This means that the CSI driver deployment will be detached from the ownership chain, that in turn will create issues for k8s to follow a proper deletion process

Reply:
https://github.com/rook/rook/blob/3b1a4286885a78737e597945cc20826a6c3137cb/pkg/operator/ceph/csi/csidriver.go#L108-L116 talks about csidriver delete logic , which can be reused here

Reply:
Discussed offline. Agreed that we need to have code in the operator to Delete/Remove the CSI driver as we cannot own cluster scoped resources and the removal is not just deletion of a resource